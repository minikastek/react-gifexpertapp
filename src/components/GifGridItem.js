import React from 'react'
import PropTypes from 'prop-types'

export const GifGridItem = ( { title, url } ) => {

    return (
        <div className="card animate__bounceIn">
            <img src={url} alt={title} />
            <p>{title}</p>
            {/* {img.title} */}
        </div>
    )
}

GifGridItem.propTypes={
    title: PropTypes.string.isRequired
}

GifGridItem.propTypes ={
    url: PropTypes.string.isRequired
}