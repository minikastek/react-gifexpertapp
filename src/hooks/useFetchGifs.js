import { useState, useEffect } from "react"
import { getGifs } from "../helpers/getGifs";

export const useFetchGifs=(category)=>{

        const [state, setState] = useState({
            data:[],
            loading: true
        });

        useEffect( ()=>{

            getGifs(category)
                .then( imgs=>{
                        setState({
                            data:imgs,
                            loading:false
                        })
                });
                
        }, [category])

        // el , [] esta hecho para que el codigo use efect solo se 
        // ejecute 1 vez, category entre los [ significa que si cambia category de 
        // alguna forma se ejecuta de nuevo]

    return state; //{data:[],loading:true};
}