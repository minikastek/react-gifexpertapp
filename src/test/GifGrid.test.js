import React from "react";
import {shallow} from 'enzyme'
import { GifGrid } from "../components/GifGrid";
import { useFetchGifs } from "../hooks/useFetchGifs";
jest.mock('../hooks/useFetchGifs');

describe('Pruebas en GifGrid',()=>{

    const category ='One Punch';

    test('prueba de snapshot ',()=>{

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={category}/>);
        expect(wrapper).toMatchSnapshot();

    });

    test('Debe de mostrar items cuando se cargan imagenes',()=>{

        const gifs =[{
            id: 'ABC',
            url: 'https://localhost/cualquier/cosa.jpg',
            title: 'cualquiero cosa'
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        
        const wrapper = shallow(<GifGrid category={category}/>);

        expect(wrapper.find('p').exists() ).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);

    })

})