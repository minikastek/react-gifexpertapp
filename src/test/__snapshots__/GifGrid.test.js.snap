// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGrid prueba de snapshot  1`] = `
<Fragment>
  <h3>
    One Punch
  </h3>
  <p>
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
